cmake/pcl_pclconfig.cmake

```diff
@@ -37,7 +37,11 @@ foreach(_ss ${PCL_SUBSYSTEMS_MODULES})
         if(_opt_deps)
             set(PCLCONFIG_OPTIONAL_DEPENDENCIES "${PCLCONFIG_OPTIONAL_DEPENDENCIES}set(pcl_${_ss}_opt_dep ")
             foreach(_opt_dep ${_opt_deps})
-                set(PCLCONFIG_OPTIONAL_DEPENDENCIES "${PCLCONFIG_OPTIONAL_DEPENDENCIES}${_opt_dep} ")
+                string(TOUPPER "WITH_${_opt_dep}" _tmp)
+                string(REGEX REPLACE "-(.*)" "" _condition ${_tmp}) #libusb-1.0 case
+                if(${${_condition}})
+                  set(PCLCONFIG_OPTIONAL_DEPENDENCIES "${PCLCONFIG_OPTIONAL_DEPENDENCIES}${_opt_dep} ")
+                endif(${${_condition}})
```

Comment:
It is not necessary to repeat condition expression in the `endif` since CMake 2.8.